src/poslenstrings.jl

```diff
@@ -0,0 +1,526 @@
+primitive type PosLen 64 end
+PosLen(x::UInt64) = Core.bitcast(PosLen, x)
+UInt64(x::PosLen) = Core.bitcast(UInt64, x)
+
+Base.convert(::Type{PosLen}, x::UInt64) = PosLen(x)
+Base.convert(::Type{UInt64}, x::PosLen) = UInt64(x)
+
+const MISSING_BIT = 0x8000000000000000
+missingvalue(x) = (UInt64(x) & MISSING_BIT) == MISSING_BIT
+
+const ESCAPE_BIT = 0x4000000000000000
+escapedvalue(x) = (UInt64(x) & ESCAPE_BIT) == ESCAPE_BIT
+
+getpos(x) = (UInt64(x) & 0x3ffffffffff00000) >> 20
```

Comment:
So this is 4TB limit on the file size. Right? Seems OK, but as below - what do we do when it is exceeded?

Reply:
We can/should add an error check in CSV.jl if an input file happens to be > 4TB and they request `lazystrings=true`.

Reply:
Ok, added check in CSV.jl here: https://github.com/JuliaData/CSV.jl/pull/837/commits/c52b8b36d27c24057ed303174991fe367eb35290